File: bridge/src/main/java/ru/hollowhorizon/hollowengine/common/registry/extend/HollowDynamicRegistry.java
package ru.hollowhorizon.hollowengine.common.registry.extend;

import net.minecraft.resources.ResourceKey;
import net.minecraft.resources.ResourceLocation;
import org.jetbrains.annotations.Nullable;

@SuppressWarnings("unchecked")
public interface HollowDynamicRegistry {
    void hollow$clearDynamic();

    default void clearDynamic() {
        hollow$clearDynamic();
    }

    @Nullable ResourceKey<?> hollow$getKey(ResourceLocation id);

    default <T> @Nullable ResourceKey<T> getKey(ResourceLocation id) {
        return (ResourceKey<T>) hollow$getKey(id);
    }

    boolean hollow$isPresent(ResourceLocation id);

    default boolean isPresent(ResourceLocation id) {
        return hollow$isPresent(id);
    }
}

File: src/main/java/ru/hollowhorizon/hollowengine/common/registry/extend/HollowDynamicRegistry.kt
package ru.hollowhorizon.hollowengine.common.registry.extend

import net.minecraft.resources.ResourceKey
import net.minecraft.resources.ResourceLocation

@Suppress("UNCHECKED_CAST")
interface HollowDynamicRegistry {
    fun `hollow$clearDynamic`()

    fun clearDynamic() = `hollow$clearDynamic`()

    fun `hollow$getKey`(id: ResourceLocation): ResourceKey<*>?

    fun <T> getKey(id: ResourceLocation): ResourceKey<T>? = `hollow$getKey`(id) as ResourceKey<T>?

    fun `hollow$isPresent`(id: ResourceLocation): Boolean

    fun isPresent(id: ResourceLocation): Boolean = `hollow$isPresent`(id)
}
